spring-pulsar-test/src/main/java/org/springframework/pulsar/test/support/Condition.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.test.support;
+
+import java.util.List;
+
+import org.apache.pulsar.client.api.Message;
+
+/**
+ * A condition to be used in {@link PulsarTestConsumer} to assert the consumed messages.
+ *
+ * @param <T> the type of the message
+ * @author Jonas Geiregat
+ */
+@FunctionalInterface
+public interface Condition<T> {
+
+	/**
+	 * Asserts that the consumed messages meet the condition.
```

Comment:
**[NIT]** Does not necessarily assert, but rather returns true if it meets the condition.